lib/CodeGen/PagerandoBinning.cpp

```diff
@@ -47,16 +41,19 @@ class PagerandoBinning : public ModulePass {
   bool runOnModule(Module &M) override;
 
   void getAnalysisUsage(AnalysisUsage &AU) const override {
+    ModulePass::getAnalysisUsage(AU);
     AU.addRequired<MachineModuleInfo>();
     AU.addPreserved<MachineModuleInfo>();
     AU.setPreservesAll();
-    ModulePass::getAnalysisUsage(AU);
   }
 
 private:
-  // Map from free space -> Bin
-  typedef std::multimap<unsigned, Bin> BinMap;
-  BinMap Bins;
+  static constexpr unsigned DefaultBin = 0;
+  static constexpr unsigned BinSize = 4096;     // one page
+  static constexpr unsigned MinFreeSpace = 64;  // cache line (32 or 64 on ARM)
```

Comment:
Does the concept of a minimum free space for a bin make sense? If a bin has less than this amount of free space remaining, we don't consider it when we look for bins for the following functions (i.e., we skip adding it to the free space mapping).